### PR TITLE
Document remote container termination safety guarantees.

### PR DESCRIPTION
**Pull request by Alexhuszagh**
Updates the comment documentation for the async-signal safety guarantees in stopping a running container.

### CODE REVIEW

**Comment by Alexhuszagh** [comment]
I double checked the entirety of `Command` for both Windows and UNIX processes to determine if there was anything potentially missing here for how we run this. There's a few minor issues that weren't documented:

**Async-Signal Unsafe**
- `Engine::drop`: `Engine` contains a `PathBuf`, which is allocated.
- `Command::new`, which creates a `CString` from an `OsStr`.
- `Command::arg`, which creates a `CString` from an `OsStr`.
- `Command::args`, which does the same as `Command::arg` for each argument.
- `DeleteContainer::drop`: Contains `Engine` and `String`, which are allocated. Also invokes `container_stop` and `container_rm`, which allocate since they create a `Command`.

**Async-Signal Safe**
- `Command::capture_envp`: since we don't modify the environment, this won't allocate.
- `Command::spawn`: as long as we don't modify the environment, `Command::capture_envp` won't allocate, and therefore neither will `spawn` as long as we're not using pipes (which we aren't).

As documented, it would be possible to make this entirely async-signal safe even without the atomic guard, however, this shouldn't be necessary.

File: src/errors.rs [hunk 1]
```diff
@@ -87,6 +87,21 @@ unsafe fn termination_handler() {
...
+    // allocation/deallocation, since we would need static global muts
+    // for basically everything. `Command::arg` and `Command::new` will
+    // internally allocate, and freeing it will deallocate any arguments
+    // it was provided. even if we used a static global `Command`, the
+    // `io::Result` requires a `Box` or `io::Error`, which would allocate.
+    // the alternative would be to use `posix_spawnp` or `CreateProcess`
+    // directly, which are async-signal safe and thread-safe, respectively,
+    // however, we'd need to store the engine path and the argument list as
+    // a global CString and `Vec<CString>`, respectively. since we have an
+    // atomic guard here, this should be safe regardless.
```
Comment (Emilgardis) [inline src/errors.rs:104]:
```suggestion
    // a global CString and `Vec<CString>`, respectively. This atomic guard makes this safe regardless.
```

**Review by Emilgardis — APPROVED** [review]
r=me on change

**Comment by Alexhuszagh** [comment]
bors r=Emilgardis

**Pull request merged**